packaging/flavorgen/cmd/root.go

```diff
@@ -52,30 +79,74 @@ func RunRoot(command *cobra.Command) error {
 	if err != nil {
 		return errors.Wrapf(err, "error accessing flag %s for command %s", flavorFlag, command.Name())
 	}
+	outputDir, err := command.Flags().GetString(outputDirFlag)
+	if err != nil {
+		return errors.Wrapf(err, "error accessing flag %s for command %s", outputDirFlag, command.Name())
+	}
+	var outputFlavors []string
+	if flavor != "" {
+		outputFlavors = append(outputFlavors, flavor)
+	} else {
+		outputFlavors = allFlavors
+	}
+	generateMultiFlavors := len(outputFlavors) > 1
+	for _, f := range outputFlavors {
+		manifest, err := generateSingle(f)
+		if err != nil {
+			return err
+		}
+
+		fileMapping, ok := flavorMappings[f]
+		if !ok {
+			return fmt.Errorf("file mapping for flavor %q is missng in flavorMappings", f)
+		}
+
+		yamlFileName := fileMapping + ".yaml"
+		if outputDir == "-" {
+			if generateMultiFlavors {
+				// use the yaml filename as a section delimiter
+				fmt.Printf("### %s\n", yamlFileName)
+			}
+			fmt.Print(manifest)
+			continue
+		}
+
+		yamlPath := filepath.Join(outputDir, yamlFileName)
+		err = os.WriteFile(yamlPath, []byte(manifest), 0600)
+		if err != nil {
+			return errors.Wrapf(err, "failed to dump manifest content to file for flavor %s", f)
```

Comment:
```suggestion
			return errors.Wrapf(err, "failed to write manifest content to file for flavor %s", f)
```
nit. Just sounds a bit nicer